x-pack/legacy/plugins/actions/server/builtin_action_types/webhook.ts

```diff
@@ -0,0 +1,154 @@
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import axios, { AxiosError, AxiosResponse } from 'axios';
+import { i18n } from '@kbn/i18n';
+import { schema, TypeOf } from '@kbn/config-schema';
+import { nullableType } from './lib/nullable';
+import { portSchema } from './lib/schemas';
+import { isOk, promiseResult, Result } from './lib/result_type';
+import { ActionType, ActionTypeExecutorOptions, ActionTypeExecutorResult } from '../types';
+
+// config definition
+enum WebhookMethods {
+  POST = 'post',
+  PUT = 'put',
+}
+enum WebhookSchemes {
+  HTTP = 'http',
+  HTTPS = 'https',
+}
+
+export type ActionTypeConfigType = TypeOf<typeof ConfigSchema>;
+
+const HeadersSchema = schema.recordOf(schema.string(), schema.string());
+
+export const CompositeUrlSchema = schema.object({
+  host: schema.string(),
+  port: portSchema(),
+  path: nullableType(schema.string()),
+  scheme: schema.oneOf(
+    [schema.literal(WebhookSchemes.HTTP), schema.literal(WebhookSchemes.HTTPS)],
+    {
+      defaultValue: WebhookSchemes.HTTP,
+    }
+  ),
+});
+
+const ConfigSchema = schema.object({
+  url: schema.oneOf([schema.string(), CompositeUrlSchema]),
+  method: schema.oneOf([schema.literal(WebhookMethods.POST), schema.literal(WebhookMethods.PUT)], {
+    defaultValue: WebhookMethods.POST,
+  }),
+  headers: nullableType(HeadersSchema),
+});
+
+// secrets definition
+export type ActionTypeSecretsType = TypeOf<typeof SecretsSchema>;
+const SecretsSchema = schema.object({
+  user: schema.string(),
+  password: schema.string(),
+});
+
+// params definition
+export type ActionParamsType = TypeOf<typeof ParamsSchema>;
+const ParamsSchema = schema.object({
+  body: schema.maybe(schema.string()),
+});
+
+// action type definition
+export const actionType: ActionType = {
+  id: '.webhook',
+  name: 'webhook',
+  validate: {
+    config: ConfigSchema,
+    secrets: SecretsSchema,
+    params: ParamsSchema,
+  },
+  executor,
+};
+
+function asComposedUrl(url: string | TypeOf<typeof CompositeUrlSchema>): string {
+  if (typeof url === 'string') {
+    return url;
+  }
+  return `${url.scheme}://${url.host}${url.port ? `:${url.port}` : ''}${url.path || ''}`;
+}
+
+function describeWebhookErrorSource(error: AxiosError): string {
+  if (error.response) {
+    // The request was made and the server responded with a status code
+    // that falls out of the range of 2xx
+    return `remote webhook failure`;
+  } else if (error.request) {
+    // The request was made but no response was received
+    return `unreachable remote webhook`;
+  }
+  return `unknown webhook event`;
+}
+
+// action executor
+async function executor(execOptions: ActionTypeExecutorOptions): Promise<ActionTypeExecutorResult> {
+  const log = (msg: string) => execOptions.services.log(['warn', 'actions', 'webhook'], msg);
```

Comment:
`warn` seems a bit loud for these.  Check the other built-in actions, I think we were trying to be fairly quiet, and even errors invoking an action would only ever be warn anyway; but we don't want to be flooding the log with tons of messages.  Maybe create a `debugLog()` which you can use liberally, and `warnLog()` for warn-level things?

Reply:
Haha I was following your example from PageDuty ;)

Simplified now to warn on error, but debug the rest of the time.

Reply:
oh!  I'll go look at PD - I might have had them artificially high during dev ...

Reply:
Oops, looking at PD now looks like you only did it for errors, but as that was where I got the reference form I reused it.

You're all good. :)